node_http_server.js

```diff
@@ -50,20 +51,34 @@ class NodeHttpServer {
       }
     });
 
-    app.use(Express.static(this.webroot));
-    app.use(Express.static(this.mediaroot));
-
     if (this.config.auth && this.config.auth.api) {
       app.use('/api/*', basicAuth(this.config.auth.api_user, this.config.auth.api_pass));
     }
     app.use('/api/streams', streamsRoute(context));
     app.use('/api/server', serverRoute(context));
 
+    if (this.config.auth !== undefined && this.config.auth.play) {
+      app.use((req, res, next) => {
+        if (!req.url.startsWith('/api')) {
```

Comment:
Here I have created new route called /media/*/*.mp4
and added below cond for verifying the token, and when I tried to access the media from the browser and pass token e.g ?sign=1503458721-80c1d1ad2e0c2ab63eebb50eed64201a but each time I'm getting Unauthorization even after passing the right token.  my aim is the person who has token only that can download the recorded video and see hls.  can you please guide what I'm doing wrong?

Reply:
you can debug NodeCoreUtils.verifyAuth and diff the token